python/phase-1.md

```diff
@@ -156,24 +181,71 @@ Notes:
   * We could make an abstract base class if we want for typing/purposes that just has the `service` property
     unimplemented
 * We call it `start_workflow` instead of just `start` to be clearer
+* We have an `execute_workflow` to combine start + result waiting
 * We have embedded `start_signal` and `start_signal_args` as parameters, so if the former is present we will perform a
   signal with start. Is this ok?
   * This is an intentional deviation from other SDKs, but makes logical sense
 * The client supports a default task queue, do we want this?
 
+### Core
+
+Exported from `temporalio` (but probably in `temporalio/core/core.py`, exported from `temporalio.core` then
+exported from `temporalio`):
+
+```python
+class Core:
+    async def create(
+        addr: str,
+        *,
+        namespace: str = "default"
+        # ... All the same options as Client.create
+    ) -> Core:
+        pass
+```
+
+Notes:
+
+* If we could use Core to back our API client, then this would not be needed
+  * Even if we didn't, maybe if the worker feature is enabled we can create a core instance during client creation? Then
+    we can pass a client to worker instead of core which would be much cleaner
+* While we could implicitly create this, say, per address I think each core instance has enough uniqueness to require
+  manual user management
+* We could flip the wording of this a bit and have a "worker" not be task queue specific and be 1:1 with core, then have
+  the concept of a "task queue worker" that then lets you register activities and such.
+  * Probably deviates too much from the norm. Still, it's quite unfortunate to have a two-step process to create a
+    worker.
+* Is there any harm in creating one core instance per worker? We could do away with this if not.
+  * Do we lose some of the shared core benefits by only registering one worker for each core instance?
+  * I assume there is a meaningful difference between 1000 core instances that we poll on each and just 1?
```

Comment:
There should only be one per namespace. There would be substantially more overhead involved in making many many more Core instances.